micromamba/tests/test_package.py

```diff
@@ -0,0 +1,209 @@
+    )
+    subprocess.call(
+        [
+            mamba_exe,
+            "package",
+            "compress",
+            str(out),
+            str(tmp_path / "mm" / "out.tar.bz2"),
+        ]
+    )
+
+    compare_two_tarfiles(
+        tarfile.open(cph_test_file), tarfile.open(tmp_path / "mm" / "out.tar.bz2")
+    )
+
+    fout = tarfile.open(tmp_path / "mm" / "out.tar.bz2")
+    names = fout.getnames()
+    assert "info/paths.json" in names
+
+    info_files = [f for f in names if f.startswith("info/")]
+    # check that info files are at the beginning
+    assert names[: len(info_files)] == info_files
+
+    # check that the rest is sorted
+    assert_sorted(names[len(info_files) :])
+    assert_sorted(names[: len(info_files)])
+
+
+def test_transmute(cph_test_file: Path, tmp_path: Path):
+    (tmp_path / "cph").mkdir(parents=True)
+    (tmp_path / "mm").mkdir(parents=True)
+
+    shutil.copy(cph_test_file, tmp_path)
+    shutil.copy(tmp_path / cph_test_file.name, tmp_path / "mm")
+
+    mamba_exe = get_umamba()
+    subprocess.call(
+        [mamba_exe, "package", "transmute", str(tmp_path / "mm" / cph_test_file.name)]
+    )
+    failed_files = cph.transmute(
+        str(tmp_path / cph_test_file.name), ".conda", out_folder=str(tmp_path / "cph")
+    )
+    assert len(failed_files) == 0
+
+    as_conda = cph_test_file.name.removesuffix(".tar.bz2") + ".conda"
+
+    cph.extract(str(tmp_path / "cph" / as_conda))
+    cph.extract(str(tmp_path / "mm" / as_conda))
+
+    conda = list((tmp_path / "cph").rglob("**/*"))
+    mamba = list((tmp_path / "mm").rglob("**/*"))
+
+    fcmp = filecmp.dircmp(
+        tmp_path / "cph" / "cph_test_data-0.0.1-0",
+        tmp_path / "mm" / "cph_test_data-0.0.1-0",
+    )
+    assert (
+        len(fcmp.left_only) == 0
+        and len(fcmp.right_only) == 0
+        and len(fcmp.diff_files) == 0
+    )
+    # fcmp.report_full_closure()
+
+    # extract zipfile
+    with zipfile.ZipFile(tmp_path / "mm" / as_conda, "r") as zip_ref:
+        l = zip_ref.namelist()
+
+        assert l[2].startswith("info-")
+        assert l[0] == "metadata.json"
+        assert l[1].startswith("pkg-")
+
+        zip_ref.extractall(tmp_path / "mm" / "zipcontents")
+
+    files = list((tmp_path / "mm" / "zipcontents").glob("**/*"))
+    for f in files:
+        if f.suffix == ".zst":
+            with open(f, mode="rb") as fi:
+                dcf = zstandard.ZstdDecompressor().stream_reader(fi)
+
+                with tarfile.open(fileobj=dcf, mode="r|") as z:
+                    assert_sorted(z.getnames())
+                    members = z.getmembers()
+                    for m in members:
+                        if f.name.startswith("info-"):
+                            assert m.name.startswith("info/")
+                        if not f.name.startswith("info-"):
+                            assert not m.name.startswith("info/")
+                        assert m.uid == 0
+                        assert m.gid == 0
```

Comment:
our new comparison is https://github.com/conda/conda-package-handling/blob/main/src/conda_package_handling/validate.py#L59

tarballs can also store uname and gname are strings

to be bug-compatible, conda-package-handling 2 keeps the "regex that puts info/LICENSE into pkg-" which we should fix in a minor 2.x release. We may audit existing packages for info/ filenames to see if there was a good reason for that regex.

Reply:
I fixed that (also setting them to empty).